code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm

```diff
@@ -56,56 +57,55 @@
 	color = "#60A584" // rgb: 96, 165, 132
 	overdose_threshold = 20
 	addiction_threshold = 10
+	metabolization_rate = 0.75* REAGENTS_METABOLISM
+	stun_threshold = 4
+	speedboost = FAST
 
 /datum/reagent/drug/crank/on_mob_life(mob/living/M)
 	var/high_message = pick("You feel jittery.", "You feel like you gotta go fast.", "You feel like you need to step it up.")
-	var/tolerance_message = pick("You feel uncomfortable.", "You feel like you're too slow.", "You feel like you don't want to go fast anymore.")
-	if(current_cycle <= 100)
-		if(prob(5))
-			M << "<span class='notice'>[high_message]</span>"
-		M.AdjustParalysis(-1)
-		M.AdjustStunned(-1)
-		M.AdjustWeakened(-1)
-	if(current_cycle > 100)
-		if(prob(15))
-			M << "<span class='notice'>[tolerance_message]</span>"
-		M.adjustStaminaLoss(2)
+	if(prob(5))
+		M << "<span class='notice'>[high_message]</span>"
+	if(!(M.stunned || M.weakened || M.paralysis))
+		stun_timer++
+		metabolization_rate = 0.75* REAGENTS_METABOLISM
+	else
+		metabolization_rate = 1.5 * REAGENTS_METABOLISM
```

Comment:
Could you not use initial(metabolization_rate) is your multiplier?
So if someone wants to change the metabolizationrate, they only need to change the value at the definitions of the reagent?